homeassistant/components/homekit/__init__.py

```diff
@@ -827,7 +819,18 @@ async def async_configure_accessories(self) -> list[State]:
                 await self._async_set_device_info_attributes(
                     ent_reg_ent, dev_reg, entity_id
                 )
-                self._async_configure_linked_sensors(ent_reg_ent, device_lookup, state)
+                if device_id := ent_reg_ent.device_id:
+                    if device_id not in device_lookup:
+                        device_lookup[device_id] = {
+                            (
+                                entry.domain,
+                                entry.device_class or entry.original_device_class,
+                            ): entry.entity_id
+                            for entry in entries.get_entries_for_device_id(device_id)
+                        }
```

Comment:
This is significantly faster in testing (almost 4x) since we only lookup devices after the entity filter